crypto-ballerina/src/crypto/tests/unit-tests/aes-cbc-encrypt-decrypt-test.bal

```diff
@@ -0,0 +1,132 @@
+        invalidKey[i] = <byte> i;
+        i = i + 1;
+    }
+    i = 0;
+    while(i < 16) {
+        iv[i] = <byte> i;
+        i = i + 1;
+    }
+    byte[]|error result = encryptAesCbc(message, invalidKey, iv, NONE);
+    if (result is error) {
+        test:assertEquals(extractErrorMessage(result), "Invalid key size. valid key sizes in bytes: [16, 24, 32]",
+            msg = "Incorrect error for invalid key while No Padding Encryption with AES CBC.");
+    } else {
+        test:assertFail(msg = "No error for invalid key while No Padding Encryption with AES CBC.");
+    }
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithAesCbcNoPaddingUsingInvalidIVLength() {
+    byte[] message = "Ballerina crypto test           ".toBytes();
+    byte[] key = [];
+    int i = 0;
+    while(i < 16) {
+        key[i] = <byte> i;
+        i = i + 1;
+    }
+    i = 0;
+    byte[] invalidIv = [];
+    while(i < 15) {
+        invalidIv[i] = <byte> i;
+        i = i + 1;
+    }
+    byte[]|error result = encryptAesCbc(message, key, invalidIv, NONE);
+    if (result is error) {
+        test:assertEquals(extractErrorMessage(result),
+            "Error occurred while AES encrypt/decrypt: Wrong IV length: must be 16 bytes long",
+            msg = "Incorrect error for invalid key while No Padding Encryption with AES CBC.");
+    } else {
+        test:assertFail(msg = "No error for invalid IV length while No Padding Encryption with AES CBC.");
+    }
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithAesCbcNoPaddingUsingInvalidInputLength() {
+    byte[] invalidMessage = "Ballerina crypto test".toBytes();
+    byte[] key = [];
+    byte[] iv = [];
+    int i = 0;
+    while(i < 16) {
+        key[i] = <byte> i;
+        i = i + 1;
+    }
+    i = 0;
+    while(i < 16) {
+        iv[i] = <byte> i;
+        i = i + 1;
+    }
+    byte[]|error result = encryptAesCbc(invalidMessage, key, iv, NONE);
+    if (result is error) {
+        test:assertEquals(extractErrorMessage(result),
+            "Error occurred while AES encrypt/decrypt: Input length not multiple of 16 bytes",
+            msg = "Incorrect error for for invalid input length while No Padding Encryption with AES CBC.");
+    } else {
+        test:assertFail(msg = "No error for invalid input length while No Padding Encryption with AES CBC.");
+    }
+}
+
+@test:Config {}
+function testEncryptAndDecryptWithAesCbcPkcs5() {
+    byte[] message = "Ballerina crypto test".toBytes();
+    byte[] key = [];
+    byte[] iv = [];
+    int i = 0;
+    while(i < 16) {
+        key[i] = <byte> i;
+        i = i + 1;
+    }
+    i = 0;
+    while(i < 16) {
+        iv[i] = <byte> i;
+        i = i + 1;
+    }
+    byte[] cipherText = checkpanic encryptAesCbc(message, key, iv, "PKCS5");
+    byte[] plainText = checkpanic decryptAesCbc(cipherText, key, iv, "PKCS5");
+    test:assertEquals(plainText.toBase16(), message.toBase16(),
+        msg = "Error while Encrypt/Decrypt with AES CBC PKCS5.");
+}
```

Comment:
The negative test case for testing encrypt and decrypt with AES CBC PKCS1 is missing here. Shall we add that also?